static/app/views/dashboardsV2/detail.tsx

```diff
@@ -293,7 +293,7 @@ class DashboardDetail extends Component<Props, State> {
     const {modifiedDashboard} = this.state;
     const newModifiedDashboard = {
       ...cloneDashboard(modifiedDashboard || dashboard),
-      widgets: widgets.map(assignTempId),
```

Comment:
I moved these to the forms when widgets get created because there were issues with `widgets` being replaced and the `tempId` was missing